### DISCUSSION TITLE
Why does "External host error causing abort - skipping" not fail the pipeline?

### DISCUSSION THREAD

**Discussion by BreakBB** [renovatebot/renovate#26559] [Request Help] [answered]
### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us what version of Renovate you run.

renovate/renovate:37.126.2@sha256:71e2cafd1e83eacbf48ad1a96aa31435e6aa3b45b1122b72314da78bc7f4bb90

### If you're self-hosting Renovate, select which platform you are using.

GitLab self-hosted

### What is your question?

I ran into a "too many requests" issue on DockerHub, which was logged as a warning. Afterwards, logged as info:

>  INFO: External host error causing abort - skipping
> DEBUG: Repository result: external-host-error, status: onboarded, enabled: true, onboarded: true

The pipline exits gracefully and announced "success", even though it did not check all dependencies. I think this is giving the wrong signal and should be a failed state.

What's the reason to mark the pipeline as being successful?

**Comment by rarkins** [comment]
Renovate exits with a non-zero exit code when _errors_ are logged. If there are temporary problems with e.g. github or docker, people generally don't want to have their error logs spammed, but perhaps we could make this configurable

**Reply by BreakBB** [reply]
Users wouldn't know about these temporary problem if they do not check the Renovate logs or start to wonder why no PRs are created.
These problems might also be permanently, if for example image mirrors go offline and one continue running into the request limit on DockerHub's free tier.

Having this at least configurable would be great 👍🏻

**Reply by BreakBB** [reply]
I guess I can not open up an issue for this, so it would be great if someone would move it into a proper feature request.

**Reply by rarkins** [reply]
Let's first decide here whether a narrow/specific feature is best (e.g. failForExternalHostError) or something more generic for remapping any log message you want (because error logs automatically cause a failed pipeline)

**Reply by BreakBB** [reply]
Well I only encountered this behavior in the context of the external host error. Not sure what other cases there might be, which also do not result in a pipeline failure, but might be interesting for _some_ users 🤔

**Reply by rarkins** [reply, marked as answer]
Created https://github.com/renovatebot/renovate/issues/26723
